cupyx/scipy/stats/_stats_py.py

```diff
@@ -0,0 +1,143 @@
+        Degrees of freedom correction in the calculation of the
+        standard deviation. Default is 0
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle when input contains nan. 'propagate'
+        returns nan, 'raise' throws an error, 'omit' performs
+        the calculations ignoring nan values. Default is
+        'propagate'. Note that when the value is 'omit',
+        nans in the input also propagate to the output,
+        but they do not affect the z-scores computed
+        for the non-nan values
+
+    Returns
+    -------
+    zscore : array-like
+        The z-scores, standardized by mean and standard deviation of
+        input array `a`
+
+    """
+
+    return zmap(a, a, axis=axis, ddof=ddof, nan_policy=nan_policy)
+
+
+def zmap(scores, compare, axis=0, ddof=0, nan_policy='propagate'):
+    """Calculate the relative z-scores.
+
+    Return an array of z-scores, i.e., scores that are standardized
+    to zero mean and unit variance, where mean and variance are
+    calculated from the comparison array.
+
+    Parameters
+    ----------
+    scores : array-like
+        The input for which z-scores are calculated
+    compare : array-like
+        The input from which the mean and standard deviation of
+        the normalization are taken; assumed to have the same
+        dimension as `scores`
+    axis : int or None, optional
+        Axis over which mean and variance of `compare` are calculated.
+        Default is 0. If None, compute over the whole array `scores`
+    ddof : int, optional
+        Degrees of freedom correction in the calculation of the
+        standard deviation. Default is 0
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle the occurrence of nans in `compare`.
+        'propagate' returns nan, 'raise' raises an exception, 'omit'
+        performs the calculations ignoring nan values. Default is
+        'propagate'. Note that when the value is 'omit', nans in `scores`
+        also propagate to the output, but they do not affect the z-scores
+        computed for the non-nan values
+
+    Returns
+    -------
+    zscore : array-like
+        Z-scores, in the same shape as `scores`
+
+    """
+
+    a = compare
+
+    if a.size == 0:
+        return cupy.empty(a.shape)
+
+    nan_policy = _contains_nan(a, nan_policy)
+
+    if nan_policy == 'omit':
+        if axis is None:
+            mn = cupy.nanmean(a.ravel())
+            std = cupy.nanstd(a.ravel(), ddof=ddof)
+            isconst = _isconst(a.ravel())
+        else:
+            mn = cupy.nanmean(a, axis=axis, keepdims=True)
+            std = cupy.nanstd(a, axis=axis, keepdims=True, ddof=ddof)
+            isconst = (_first(a, axis) == a).all(axis=axis, keepdims=True)
+    else:
+        mn = a.mean(axis=axis, keepdims=True)
+        std = a.std(axis=axis, ddof=ddof, keepdims=True)
+        if axis is None:
+            isconst = (a.item(0) == a).all()
```

Comment:
`ndarray.item()` transfers data from GPU to the host, then convert it to a Python scalar. We may use `a.ravel()[0]` instead to make a comparison with `a` whose data is on GPU.

Reply:
Got that! Thanks for the concept :)